### PR TITLE
Bug 1120739 Add Hello snippet with animation

### PR DESCRIPTION
**Pull request by sgarrity**
[Bug 1120739](https://bugzilla.mozilla.org/show_bug.cgi?id=1120739)
Test here: https://snippets.allizom.org/show/69/

### CODE REVIEW

File: templates/hello/animation-snippet.html [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      brandLogo.style.backgroundImage = 'url(' + animationImg.src + ')';
+      brandLogo.className += ' switch-{{ snippet_id }}';
+      brandLogo.appendChild(snippetContents);
+      setTimeout(function() {
+        brandLogo.className += ' show-contents-{{ snippet_id }}';
+      }, contentDelay);
+    });
+
+
+    // Begin UITour shinanigins
```
Comment (sgarrity) [inline templates/hello/animation-snippet.html]: Most of the code below was taken directly, or slightly modified, from this page: https://github.com/mozilla/snippets/blob/master/templates/hello/simple-snippet.html

**Comment by ckprice** [comment]
r+ on all of the UITour Javascript interaction (line 85+).

@glogiotatidis are you comfortable reviewing the CSS and loading of the animation?

**Comment by glogiotatidis** [comment]
Sure I'll review tomorrow.

**Comment by ckprice** [comment]
Thanks Giorgos! Let me know if you run into any issues. Alex said he can help too.

**Comment by ckprice** [comment]
@glogiotatidis please hold on code review. There could be some late changes coming in from PMM.

**Comment by glogiotatidis** [comment]
@ckprice ok, ping again me when ready

File: templates/hello/animation-snippet.html [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  var snippet = document.getElementById('snippet-{{ snippet_id }}');
+
+  snippet.addEventListener('show_snippet', function() {
+
+    var snippetContents = document.getElementById('snippet-{{ snippet_id }}-contents');
+    var contentDelay = 10000; // Delay before showing text/button
+
+    // Preload the new animation image from the CDN 
+    var animationImg = new Image();
+    animationImg.src = 'https://snippets.allizom.org/media/files/08f669b5-5e85-4b68-8025-d27f76913ad0.gif';
```
Comment (glogiotatidis) [inline templates/hello/animation-snippet.html]: Prod url https://snippets.cdn.mozilla.net/media/files/cf92d5d5-3ae5-434d-9b23-ba9ab63a2fd0.gif

**Comment by glogiotatidis** [comment]
ckprice: "Also let's add 5px of top/bottom padding around the copy block"

**Comment by glogiotatidis** [comment]
This one has the wrong bug number.

**Comment by sgarrity** [comment]
> ckprice: "Also let's add 5px of top/bottom padding around the copy block"

@ckprice Can you be more specific about this? We're fairly limited in height.

**Comment by ckprice** [comment]
@sgarrity this was a P2 request from Fabio. If we are limited in vertical space, please ignore.

**Comment by sgarrity** [comment]
I've tried switching back from the CDN to a directly embedded image, but I get a half-second(ish) delay when loading.

If I understand the snippets correctly, I can't hide the Firefox logo by default (until the snippet JS runs and adds a custom CSS class) because the snippet code needs to be completely self-contained.

**Comment by ckprice** [comment]
Summoning @Osmose...

1. Once the CDN was made available to us, my understanding is we should use it to externally load graphics as the default. As opposed to embedding the image into the JS. Mike can you confirm which should be the default?
2. Mike, any recommendation on how to hide the static logo? Showing the Firefox logo for a brief glimpse, then showing the Hello animation looked a bit glitchy. But if this is best practice, we should go with it.

Thanks

**Comment by Osmose** [comment]
> Once the CDN was made available to us, my understanding is we should use it to externally load graphics as the default. As opposed to embedding the image into the JS. Mike can you confirm which should be the default?

I don't think there is one default, they have different uses. Relying on the CDN is useful for large images that bloat the snippet filesize and that would get some benefit out of being cached across locales, whereas embedding the image into the snippet is useful when you want an image to be available immediately when the snippet code executes instead of having to deal with the separate load of an external resource.

> Mike, any recommendation on how to hide the static logo? Showing the Firefox logo for a brief glimpse, then showing the Hello animation looked a bit glitchy. But if this is best practice, we should go with it.

Assume that it is unavoidable to have the Firefox logo visible before the snippet loads. You can make it nice by fading it out instead of just directly replacing it, but you can't get rid of that gap (as the page is already displayed by the time snippet code starts executing).

**Comment by ckprice** [comment]
Thanks Mike!

@glogiotatidis Fabio has asked we bring what we have to production. Can this be ready for him in its current state Monday?

**Comment by glogiotatidis** [comment]
The snippet that we serve is different from the one in this PR. @sgarrity can you please update?

**Comment by glogiotatidis** [comment]
I reviewed the code in the template https://snippets.mozilla.com/admin/base/snippettemplate/31/ and I found one issue:

On page re-load we don't rewind the animation. Instead the animation continues (or if it's done displays the very last frame). I can reproduce this on about:home environment 

Other than that the snippet works fine, nice work @sgarrity

**Comment by sgarrity** [comment]
> The snippet that we serve is different from the one in this PR. @sgarrity can you please update?

Where did you get the version you're using? My intention is to use the version in the PR.

**Comment by glogiotatidis** [comment]
Merged both cdn and embeded versions https://github.com/mozilla/snippets/commit/569da43a31c7a91cc2c7fbbcbc03484b303c22e5  https://github.com/mozilla/snippets/commit/f9be45856cdf18bfed34e8bdc49ecbb049568245 https://github.com/mozilla/snippets/commit/8334f4995ad95880289f1e02d31c91c61620e51c

**Comment by glogiotatidis** [comment]
Thanks @sgarrity !

**Pull request closed without merging**